### PR TITLE
Improve performance of H&S flow operations: ISL bandwidth updates

### PR DESCRIPTION
**Pull request by sergii-iakovenko**
The changes:
- Optimize updates of ISL available bandwidth. This is preparation for next parallelism improvements.

### CODE REVIEW

**Review by dpoltavets — APPROVED** [review]

File: src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/common/actions/BaseFlowPathRemovalAction.java [hunk 1]
```diff
@@ -57,11 +57,8 @@ protected void updateIslsForFlowPath(FlowPath... paths) {
     }
 
     private void updateAvailableBandwidth(SwitchId srcSwitch, int srcPort, SwitchId dstSwitch, int dstPort) {
-        long usedBandwidth = flowPathRepository.getUsedBandwidthBetweenEndpoints(srcSwitch, srcPort,
-                dstSwitch, dstPort);
-        log.debug("Updating ISL {}_{} - {}_{} with used bandwidth {}", srcSwitch, srcPort, dstSwitch, dstPort,
-                usedBandwidth);
-        islRepository.updateAvailableBandwidth(srcSwitch, srcPort, dstSwitch, dstPort, usedBandwidth);
+        log.debug("Updating ISL {}_{} - {}_{} with used bandwidth", srcSwitch, srcPort, dstSwitch, dstPort);
```
Comment (niksv) [inline src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/common/actions/BaseFlowPathRemovalAction.java:60]: `with used bandwidth"` 
There is no used bandwidth in log now

File: src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/common/actions/BaseResourceAllocationAction.java [hunk 2]
```diff
@@ -227,53 +232,48 @@ protected FlowPathPair createFlowPathPair(Flow flow, List<FlowPath> pathsToReuse
...
-        if (!forceToIgnoreBandwidth && (islAvailableBandwidth + allowedOverprovisionedBandwidth) < 0) {
-            throw new ResourceAllocationException(format("ISL %s_%d-%s_%d was overprovisioned",
-                    srcSwitch, srcPort, dstSwitch, dstPort));
+            return result;
+        });
+        // Update ISLs which are occupied by the path.
+        Map<IslEndpoints, Long> updatedIsls = islRepository.updateAvailableBandwidthOnIslsOccupiedByPath(pathId);
+        for (Entry<IslEndpoints, Long> entry : updatedIsls.entrySet()) {
+            IslEndpoints isl = entry.getKey();
+            log.debug("ISL {}_{}-{}_{} was updated with used bandwidth",
```
Comment (niksv) [inline src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/common/actions/BaseResourceAllocationAction.java:266]: there is no used bw in log

File: src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/create/action/ResourcesAllocationAction.java [hunk 3]
```diff
@@ -310,26 +312,16 @@ private void allocateProtectedPath(FlowCreateFsm stateMachine) throws Unroutable
...
-        long islAvailableBandwidth =
-                islRepository.updateAvailableBandwidth(srcSwitch, srcPort, dstSwitch, dstPort, usedBandwidth);
-        if (islAvailableBandwidth < 0) {
-            throw new ResourceAllocationException(format("ISL %s_%d-%s_%d was overprovisioned",
-                    srcSwitch, srcPort, dstSwitch, dstPort));
+    private void updateIslsForFlowPath(PathId pathId) throws ResourceAllocationException {
+        Map<IslEndpoints, Long> updatedIsls = islRepository.updateAvailableBandwidthOnIslsOccupiedByPath(pathId);
+        for (Entry<IslEndpoints, Long> entry : updatedIsls.entrySet()) {
+            IslEndpoints isl = entry.getKey();
+            log.debug("ISL {}_{}-{}_{} was updated with used bandwidth",
```
Comment (niksv) [inline src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/create/action/ResourcesAllocationAction.java:319]: same

File: src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/create/action/ResourcesDeallocationAction.java [hunk 4]
```diff
@@ -95,10 +95,7 @@ private void updateIslsForSegments(List<PathSegment> pathSegments) {
     }
 
     private void updateAvailableBandwidth(SwitchId srcSwitch, int srcPort, SwitchId dstSwitch, int dstPort) {
-        long usedBandwidth = flowPathRepository.getUsedBandwidthBetweenEndpoints(srcSwitch, srcPort,
-                dstSwitch, dstPort);
-        log.debug("Updating ISL {}_{} - {}_{} with used bandwidth {}", srcSwitch, srcPort, dstSwitch, dstPort,
-                usedBandwidth);
-        islRepository.updateAvailableBandwidth(srcSwitch, srcPort, dstSwitch, dstPort, usedBandwidth);
+        log.debug("Updating ISL {}_{} - {}_{} with used bandwidth", srcSwitch, srcPort, dstSwitch, dstPort);
```
Comment (niksv) [inline src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/create/action/ResourcesDeallocationAction.java:98]: same

File: src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/common/actions/BaseFlowPathRemovalAction.java (hunk 1 above)
Comment (sergii-iakovenko) [reply src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/common/actions/BaseFlowPathRemovalAction.java]: Right, the correct log message is in IslRepository. This excessive one has been removed.

File: src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/common/actions/BaseResourceAllocationAction.java (hunk 2 above)
Comment (sergii-iakovenko) [reply src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/common/actions/BaseResourceAllocationAction.java]: Right, the correct log message is in IslRepository. This excessive one has been removed.

File: src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/create/action/ResourcesAllocationAction.java (hunk 3 above)
Comment (sergii-iakovenko) [reply src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/create/action/ResourcesAllocationAction.java]: Right, the correct log message is in IslRepository. This excessive one has been removed.

File: src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/create/action/ResourcesDeallocationAction.java (hunk 4 above)
Comment (sergii-iakovenko) [reply src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/create/action/ResourcesDeallocationAction.java]: Right, the correct log message is in IslRepository. This excessive one has been removed.

**Review by surabujin — APPROVED** [review]

**Pull request merged**